cardano-client/src/Cardano/Client/API2.hs

```diff
@@ -0,0 +1,318 @@
+-- consensus uses byron types
+-- type-instantiations for codecs
+
+module Cardano.Client.API
+(
+-- Ouroboros.Network.Point
+      WithOrigin (..)
+    , blockPointHash
+    , blockPointSlot
+    , block
+    , fromWithOrigin
+-- Ouroboros.Network.Block
+    , Point (..)
+    , pointSlot
+    , SlotNo (..)
+--     , Tip (Tip, tipPoint, tipBlockNo)
+    , BlockNo(unBlockNo, BlockNo)
+    , blockPoint
+    , decodePoint
+    , encodePoint
+    , genesisPoint
+--    , genesisBlockNo
+    , encodeTip
+    , decodeTip
+    , ChainHash (..)
+--    , genesisSlotNo
+  -- Ouroboros.Network.Mux
+    , AppType (..)
+    , OuroborosApplication (..)
+-- Ouroboros.Network.NodeToClient
+    , ErrorPolicyTrace (..)
+--    , IPSubscriptionTarget (..)
+    , LocalAddresses (..)
+    , NodeToClientProtocols (..)
+--    , NetworkIPSubscriptionTracers (..)
+    , NodeToClientVersionData (..)
+--    , SubscriptionParams (..)
+    , WithAddr (..)
+    , ncSubscriptionWorker_V1
+    , networkErrorPolicies
+    , newNetworkMutableState
+    , nodeToClientCodecCBORTerm
+-- Ouroboros.Network.Protocol.ChainSync.ClientPipelined
+--    , ChainSyncClientPipelined (..)
+--  , ClientPipelinedStIdle (..)
+--   , ClientPipelinedStIdle  ( SendMsgFindIntersect
+--                           , SendMsgRequestNext
+--                           , SendMsgRequestNextPipelined
+--                           , CollectResponse)
+    , ClientStNext (..)
+--    , chainSyncClientPeerPipelined -- cardano-explorer
+--    , ClientPipelinedStIntersect (..)
+-- Ouroboros.Network.Protocol.ChainSync.PipelineDecision
+    , pipelineDecisionLowHighMark
+    , PipelineDecision (..)
+    , runPipelineDecision
+    , MkPipelineDecision
+-- Ouroboros.Network.Protocol.ChainSync.Codec
+    , codecChainSync
+-- Ouroboros.Network.Protocol.ChainSync.Type
+    , ChainSync
+-- Ouroboros.Network.Protocol.LocalTxSubmission.Client
+    , LocalTxSubmissionClient (..)
+    , LocalTxClientStIdle (SendMsgSubmitTx)
+    , localTxSubmissionClientPeer
+-- Ouroboros.Network.Protocol.LocalTxSubmission.Codec
+    , codecLocalTxSubmission
+-- Ouroboros.Network.Protocol.LocalTxSubmission.Type
+    , LocalTxSubmission
+-- Network.TypedProtocol.Codec
+    , Codec
+-- Network.TypedProtocol.Codec.Cbor
+--    , DeserialiseFailure
+-- Network.TypedProtocol.Driver
+    , runPeer
+    , runPipelinedPeer
+-- Network.TypedProtocol.Pipelined
+    , Nat(Zero, Succ)
+-- Ouroboros.Consensus.Ledger.Byron
+    , ByronBlock (ByronBlock, byronBlockRaw, byronBlockSlotNo, byronBlockHash)
```

Comment:
If we want to export `Byron`, `Shelley, `Gougen`  I think it would be better to export each from its own module, and then re-export them at the top level. It would give some structure to the client library.

Reply:
Maybe this is also a good idea for protocols: export all the `client` protocols from `Cardano.Client.Protocols` first and then from the top level module (`Cardano.Client`).